src/app/workflow/executions/executions-tab.component.ts

```diff
@@ -0,0 +1,184 @@
+import { SessionQuery } from '../../shared/session/session.query';
+import { takeUntil } from 'rxjs/operators';
+import { BioWorkflow, Notebook, Service } from '../../shared/swagger';
+import { CheckerWorkflowService } from '../../shared/state/checker-workflow.service';
+import { CheckerWorkflowQuery } from '../../shared/state/checker-workflow.query';
+import PartnerEnum = CloudInstance.PartnerEnum;
+import { MatSelectChange } from '@angular/material/select';
+
+@Component({
+  selector: 'app-executions-tab',
+  templateUrl: './executions-tab.component.html',
+  styleUrls: ['./executions-tab.component.css'],
+})
+export class ExecutionsTabComponent extends EntryTab implements OnChanges {
+  metrics: Map<PartnerEnum, Metrics>;
+  trsID: string;
+  currentPartner: PartnerEnum;
+  partners: PartnerEnum[];
+  metricsTable: any[];
+  validationsTable: any[];
+  metricsColumns: string[];
+  validationsColumns: string[];
+  totalExecutions: number;
+  successfulExecutions: number;
+  failedExecutions: number;
+  ifMetricsExist: boolean;
+
+  @Input() entry: BioWorkflow | Service | Notebook;
+  @Input() version: WorkflowVersion;
+
+  constructor(
+    private extendedGA4GHService: ExtendedGA4GHService,
+    private workflowQuery: WorkflowQuery,
+    protected sessionQuery: SessionQuery,
+    private checkerWorkflowService: CheckerWorkflowService,
+    private checkerWorkflowQuery: CheckerWorkflowQuery
+  ) {
+    super();
+  }
+
+  ngOnChanges() {
+    this.ifMetricsExist = false;
+    this.trsID = this.checkerWorkflowQuery.getTRSId(this.entry);
+    this.metrics = new Map();
+    this.metricsTable = [];
+    this.validationsTable = [];
+    this.partners = [];
+    this.extendedGA4GHService
+      .aggregatedMetricsGet(this.trsID, this.version.name)
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe((metrics) => {
+        if (metrics) {
+          for (const [partner, metric] of Object.entries(metrics)) {
+            this.setMetricsObject(this.metrics, partner, metric);
+          }
+          this.partners = Array.from(this.metrics.keys());
+          if (this.partners.length > 0) {
+            this.ifMetricsExist = true;
+            this.selectPartner(this.partners[0]);
+          }
+        }
+      });
+  }
+
+  private setMetricsObject(metricsMap: Map<PartnerEnum, Metrics>, partner: string, metric: Metrics) {
+    for (const partnerEnum in PartnerEnum) {
+      if (PartnerEnum[partnerEnum] === partner) {
+        metricsMap.set(PartnerEnum[partnerEnum], metric);
+      }
+    }
+  }
+
+  private loadMetricsData(partner: PartnerEnum) {
+    this.metricsTable = [];
+    const metrics = this.metrics.get(partner);
+    if (metrics) {
+      this.metricsColumns = ['metric', 'min', 'avg', 'max'];
+      this.insertToMetricsTable(this.metricsTable, metrics?.cpu, 'CPU');
+      this.insertToMetricsTable(this.metricsTable, metrics?.memory, 'Memory');
+      this.insertToMetricsTable(this.metricsTable, metrics?.executionTime, 'Run Time (s)');
```

Comment:
I assume the `(s)` in `Run Time (s)` is to indicate that it's in seconds. I think it might be more clear to display next to the numerical values like `10s`.

Also, memory has a unit as well, GB. You can retrieve the unit using `metrics?.memory?.unit` and `metrics?.executionTime?.unit`

Reply:
I updated the screenshots in the PR description to show this change. Thanks for the suggestion!